include/oneapi/internal/types.h

```diff
@@ -49,6 +49,20 @@ services::String getKeyFPType()
     return services::String();
 }
 
+template<typename algorithmFPType>
+services::String getKeyType(services::String typeName)
+{
+    if (IsSameType<algorithmFPType, float>::value)
```

Comment:
Shouldn't `float` be `uint`? If it is intended behaviour (get int type which can store floating point), name this method more specifically and move to gradient boosting, since it does not look very common. I recommend to split this method into:

1. Type-trait structure
    ```cpp
    template <typename FPType>
    struct GetIntegerTypeForFPType;

    template <>
    struct GetIntegerTypeForFPType<float> {
        using Type = std::uin32_t;
    };

    template <>
    struct GetIntegerTypeForFPType<double> {
        using Type = std::uin64_t;
    };
    ```

2.  Method with straightforward logic that would form OpenCL compilation options.
    ```cpp
    template <typename algorithmFPType>
    services::String getOpenCLKeyType(const services::String &typeName);

    template <>
    inline services::String getOpenCLKeyType<std::uint32_t>(const services::String &typeName) {
        return services::String(" -D ") + typeName + services::String("=uint ");
    }

    // ...
    ```